### PR TITLE
[FIX] account: account: fiscal pos differents bill/delivery addresses

### PR DESCRIPTION
**Pull request by guva-odoo**
Little improvement of bcaf8cb859dc2f7e975d180ba0fc49da20204557
as we forgot some use cases

opw-4072691

### CODE REVIEW

**Review by epictete — COMMENTED** [review]
It was again becoming very confusing so I tried to make it a bit easier to understand.
Please let me know what you think.

File: addons/account/models/partner.py [hunk 1]
```diff
@@ -240,27 +240,26 @@ def _get_fiscal_position(self, partner, delivery=None):
         company = self.env.company
         eu_country_codes = set(self.env.ref('base.europe').country_ids.mapped('code'))
```
Comment (epictete) [inline addons/account/models/partner.py:241]: Sorry, I forgot to ask earlier, but while we're at it, shouldn't we also use `_xmlid_to_res_model_res_id()` here to reduce the number of queries?

Comment (guva-odoo) [reply addons/account/models/partner.py]: Done, I also reverted the commit about the query count in tests

**Review by epictete — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by wtaferner** [comment]
@epictete @guva-odoo 
FYI, this is not correct if the invoicing partner is B2B and has no VAT number, but yeah, I know there are many more use cases missing which are "out of scope". Not sure if these fixes do any good in a stable version to be honest.

**Comment by tsb-odoo** [comment]
> @epictete @guva-odoo FYI, this is not correct if the invoicing partner is B2B and has no VAT number, but yeah, I know there are many more use cases missing which are "out of scope". Not sure if these fixes do any good in a stable version to be honest.
Thanks for the feedback. We could include the case where the invoicing partner is B2B and has no VAT number by checking if the partner is an individual or a company, probably. But as you said, fixes in stable can do no good. I wonder if diving deeper in the nuances is really worth it. Don't hesitate to share your thoughts about that, by email if you want, and we'll look into it with pleasure. Thank you for your watchfulness.

**Pull request closed without merging**